purchase_security/views/res_partner_views.xml

```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="view_res_partner_filter" model="ir.ui.view">
+        <field name="name">res.partner.select</field>
+        <field name="model">res.partner</field>
+        <field name="inherit_id" ref="base.view_res_partner_filter" />
+        <field name="groups_id" eval="[(4, ref('group_purchase_own_orders'))]" />
+        <field name="arch" type="xml">
+            <field name="user_id" position="after">
```

Comment:
This should go better on the purchase part of the page.

Reply:
In the form view everything is added in the "Purchase" part, but the `purchase_user_id` field is also added in the tree and search views.